### PR TITLE
Fix HashMap put/remove returned value for empty key after clear.

### PR DESCRIPTION
**Pull request by bruno-roustant**
While finishing to test the small fork of HPPC in Lucene, I discovered a small bug in HashMap put and remove, in the returned value, after clear was called on the map. This is a edge case clearly, but I found a bug after 14 years of HPPC :)

This PR also adds more tests, making the randomized test against JDK HashMap run even for primitive keys/values (that revealed the small bug).

### CODE REVIEW

File: hppc/src/test/templates/com/carrotsearch/hppc/KTypeHashSetTest.java [hunk 1]
```diff
@@ -82,11 +84,14 @@ public void testIndexMethods()
 
       Assertions.assertThat(set.indexGet(set.indexOf(keyE))).isEqualTo(keyE);
       Assertions.assertThat(set.indexGet(set.indexOf(key1))).isEqualTo(key1);
+      boolean expectedExceptionThrown = false;
       try {
         set.indexGet(set.indexOf(key2));
-        fail();
```
Comment (bruno-roustant) [inline hppc/src/test/templates/com/carrotsearch/hppc/KTypeHashSetTest.java:87]: This fail would be caught anyway. Fix it.

File: hppc/src/test/templates/com/carrotsearch/hppc/KTypeHashSetTest.java [hunk 2]
```diff
@@ -393,16 +421,15 @@ public void testNullKey()
     }
     /*! #end !*/
     
-    /*! #if ($TemplateOptions.KTypeGeneric) !*/
```
Comment (bruno-roustant) [inline hppc/src/test/templates/com/carrotsearch/hppc/KTypeHashSetTest.java:396]: Make this randomized test run for primitive keys/values.

**Review by dweiss — APPROVED** [review]
Oh boy... What shocks me more is not that you've found a bug - these are always there - but that it's been 14 years (!)... 

Thank you!

**Review by dweiss — COMMENTED** [review]
Thanks, @bruno-roustant ! I'll do a few other cleanups and I think this warrants a release!
